keystone-shell/keystone_shell_vars.py

```diff
@@ -0,0 +1,152 @@
+from requests import HTTPError
+
+from kubernetes import KubernetesAPIClient, KubernetesAPIResponse
+
+NAMESPACE_FILE = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
+
+LOG_LEVEL = logging.getLevelName(os.environ.get('LOG_LEVEL', 'INFO'))
+logging.basicConfig(level=LOG_LEVEL,
+                    handlers=(logging.StreamHandler(sys.stderr),))
+
+logger = logging.getLogger(__name__)
+
+KEYSTONE_PASSWORD_ARGS = [
+    'auth_url', 'username', 'password', 'user_id', 'user_domain_id',
+    'user_domain_name', 'project_id', 'project_name', 'project_domain_id',
+    'project_domain_name', 'tenant_id', 'tenant_name', 'domain_id',
+    'domain_name', 'trust_id', 'default_domain_id', 'default_domain_name'
+]
+KEYSTONE_TIMEOUT = int(os.environ.get('KEYSTONE_TIMEOUT', '10'))
+KEYSTONE_VERIFY = os.environ.get('KEYSTONE_VERIFY', 'true') == 'true'
+KEYSTONE_CERT = os.environ.get('KEYSTONE_CERT', None)
+
+_kubernetes_client = None
+
+
+def get_current_namespace() -> str:
+    if 'NAMESPACE' in os.environ:
+        return os.environ['NAMESPACE']
+
+    if os.path.exists(NAMESPACE_FILE):
+        with open(NAMESPACE_FILE, 'r') as f:
+            return f.read()
+
+    logger.warning('Not running in cluster and $NAMESPACE is not set, '
+                   'assuming \'default\'!')
+    return 'default'
+
+
+def get_kubernetes_client() -> KubernetesAPIClient:
+    global _kubernetes_client
+
+    if _kubernetes_client is None:
+        _kubernetes_client = KubernetesAPIClient()
+        _kubernetes_client.load_auto_config()
+
+    return _kubernetes_client
+
+
+def keystone_env_vars():
+    ret = {}
+    for arg in KEYSTONE_PASSWORD_ARGS:
+        name = 'OS_{}'.format(arg.upper())
+        if name in os.environ:
+            ret[name] = os.environ[name]
+
+    return ret
+
+
+def keystone_args_from_env():
+    ret = {}
+    for arg in KEYSTONE_PASSWORD_ARGS:
+        ret[arg] = os.environ.get('OS_{}'.format(arg.upper()))
+
+    return ret
+
+
+def get_keystone_client():
+    auth = Password(**keystone_args_from_env())
+    session = Session(auth=auth,
+                      app_name='keystone-shell',
+                      user_agent='keystone-shell',
+                      timeout=KEYSTONE_TIMEOUT,
+                      verify=KEYSTONE_VERIFY,
+                      cert=KEYSTONE_CERT)
+
+    discover = Discover(session=session)
+    return discover.create_client()
+
+
+def get_kubernetes_secret(name: str,
+                          namespace: str=None) -> Union[KubernetesAPIResponse,
+                                                        None]:
+    """
+    :return: loaded secret dict or None if it does not exist
+    """
+    client = get_kubernetes_client()
+
+    if namespace is None:
+        namespace = get_current_namespace()
+
+    try:
+        return client.get('/api/v1/namespaces/{}/secrets/{}', namespace, name)
+    except HTTPError as e:
+        if e.response.status_code != 404:
+            raise
+
+        return None
```

Comment:
I think this return should have one less indentation level. I know you run it from `except` but this way you will make it default.

Reply:
Not that it matters a ton since the code paths are effectively equivalent but my feeling was that (without the conditional to check for a 404) is a bit more clear about the intent.

Ideally it would've been:
```
try:
    return client.get(...)
except NotFound:
    return None
````

... which I think makes the intent more clear - that is, `None` isn't the "default" branch so much as the "not found" branch. Since requests doesn't have exception subclasses for different error codes, it gets a bit uglier when we compare the error code ... I'm not sure how much more readable the end result is, though.